Add support for <colgroup>, <tfoot>,  and more table and cell attributes

djerius opened the pull request:
- `<table>` element:

   - add direct specification of 'id', 'class', and 'style' attributes
     via the 'id', 'class', and 'style' parameters.

   - Add generic attributes via the 'attr' parameter

   - add `<colgroup>` and `<tfoot>` elements

- table cells:

   - Add 'class', 'header', 'id', 'scope', 'style' attributes

   - allow specification of the type of cell (e.g. 'th' or 'td')

perlancar commented:
Sorry for the delayed response.

Some comments:

1. I prefer good ol' die instead of croak. When one wants to see a stack trace, one can use Devel::Confess.
2. May I suggest renaming the attribute 'tag' to 'html_element' and terminology in the docs, some tags in the docs should perhaps be more appropriately written as (HTML) element.
3. Other attributes that are HTML-specific should perhaps be prefixed with "html_", so interoperability with other Text::Table:: is easier.

djerius commented:
Will do. Thanks for the feedback.

djerius commented:
Hi, 

Sorry for the delay! I've modified the patch per your suggestions.

I didn't change the `caption` option to `html_caption` as the ability to add a caption to a table isn't HTML specific.

I've also added support for the `separate_rows` option present in [Text::Table::More](https://metacpan.org/pod/Text::Table::More)

perlancar commented:
Thanks! Merged and released 0.001.
